lib/doctest.js

```diff
@@ -23,27 +24,32 @@ function inferType(path) {
   switch (pathlib.extname (path)) {
     case '.coffee': return 'coffee';
     case '.js':     return 'js';
+    case '.mjs':    return 'esm';
     default:        throw new Error ('Cannot infer type from extension');
   }
 }
 
-var rewriters = {coffee: rewrite$coffee, js: rewrite$js};
+var rewriters = {coffee: rewrite$coffee, js: rewrite$js, esm: rewrite$js};
 
 function evaluate(moduleType, source, path) {
   return moduleType === 'commonjs' ?
     commonjsEval (source, path) :
+    moduleType === 'esm' ?
+    esmEval(source, path) :
     functionEval (source);
 }
 
 module.exports = function(path, options) {
   if (options.module != null &&
       options.module !== 'amd' &&
-      options.module !== 'commonjs') {
+      options.module !== 'commonjs' &&
+      options.module !== 'esm') {
     throw new Error ('Invalid module `' + options.module + "'");
   }
   if (options.type != null &&
       options.type !== 'coffee' &&
-      options.type !== 'js') {
+      options.type !== 'js' &&
+      options.type !== 'esm') {
```

Comment:
ESM is kind of like a "module system" as well as a "language feature", so I think it's valid to have it be an `options.module` *and* an `options.type`.